lib/ex_phone_number/metadata.ex

```diff
@@ -134,11 +134,20 @@ defmodule ExPhoneNumber.Metadata do
       if length(regions) == 1 do
         Enum.at(regions, 0)
       else
+        regions = if_gb_regions_ensure_gb_first(regions)
         get_region_code_for_number_from_region_list(phone_number, regions)
       end
     end
   end
 
+  # Ensure `GB` is first when checking numbers that match `country_code: 44`. In the Javascript official library it's the case.
+  defp if_gb_regions_ensure_gb_first(regions) do
```

Comment:
I'd say that it would be cool if `PhoneNumberMetadataForTesting.xml` was fixed in original lib as well. Then it would be a no brainer 😄 Makes me wonder how those JS tests pass after all. Could it be that something else is making correct country be used? Can you link those tests or something, I'd like to verify how JS behaves, maybe it's a PR waiting to happen for original lib. Thanks!

Reply:
Simply, the `countryCodeToRegionCodeMap` in JS is hardcoded, and `GB` is before `GG` : https://github.com/googlei18n/libphonenumber/blob/c3afee5ce9d7cd5d787ae0231b8ec41d1316f50c/javascript/i18n/phonenumbers/metadatafortesting.js#L39

We're building our `country_code_to_region_code_map` from reading the file - thus we can't hardcode the order we want for each case. My function kinda "hardcode" GB being first. Please, make me happy and tell me you have a better idea :D

Reply:
Nope, no better idea unfortunately. Let's leave it like that 👍